src/client/content_mapblock.cpp

```diff
@@ -969,35 +992,42 @@ void MapblockMeshGenerator::drawPlantlike()
 		break;
 	}
 
+	int o;
+	if (p.Y % 2 == 0) {
+		o = 1;
+	} else {
+		o = -1;
+	}
```

Comment:
Surely variation according to X and Z is required too to avoid side-by-side z-fighting, because the angles have returned to 45 degrees?